### DISCUSSION TITLE
Finding optimal height of box to fit an enumerate list in without any overlapping

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#3156] [Q&A]
Consider the following code, whose output is displayed below. Obviously, some rectangles are overlapped in the output, as the box is smaller of the height of some rectangles. 

My question is: Is there some automatic way to calculate the minimum height of the box that does not lead to any overlapping of rectangles?

```
#box(height: 50pt,
 columns(4, gutter: 11pt)[
  #enum(
    enum.item([
      #rect(width: 30pt, height: 30pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 40pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 30pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 80pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 50pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 30pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 40pt)
    ]),
    enum.item([
      #rect(width: 30pt, height: 30pt)
    ])),
])
```

![image](https://github.com/typst/typst/assets/75945439/ee3a3431-1ffa-47fe-b574-50b59bbae134)

**Comment by PgBiel** [comment]
You can use the `measure(content, styles)` function to find out approximately how tall some content will be. You can use that in your favor in the case of enums, though note that it isn't very good e.g. with paragraphs of text, since in principle text can always occupy a single line (until placed in a more restrictive environment).

Although, if you don't care too much about how tall the enum will be, you can also just give it a lot of space, like 1000pt or something.

With that said, enums, just like lists, tables and grids, will have that overlapping effect in many unexpected situations, and that is tracked in https://github.com/typst/typst/issues/3148. If that is fixed in time for the next update, it is possible that you won't have to take any further action.

**Comment by psads-git** [comment]
The `measure` idea is a good one, @PgBiel. Thanks! I will try with it.
